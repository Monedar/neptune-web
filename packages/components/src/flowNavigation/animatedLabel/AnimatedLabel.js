import React from 'react';
import Types from 'prop-types';
import CSSTransition from 'react-transition-group/CSSTransition';
import classNames from 'classnames';

import './AnimatedLabel.css';

const AnimatedLabel = ({ activeStep, backward, labels }) => {
  return labels.map((label, key) => (
    <CSSTransition
      // eslint-disable-next-line
      key={key}
      in={key === activeStep - 1}
      timeout={450}
      classNames="np-animated-label"
      unmountOnExit
    >
      <div
        className={classNames(`np-animated-label`, { 'np-animated-label--backward': backward })}
        aria-hidden={key !== activeStep - 1}
      >
        {label}
      </div>
    </CSSTransition>
  ));
};

/* eslint-enable react/no-array-index-key */

AnimatedLabel.propTypes = {
  labels: Types.arrayOf(Types.node).isRequired,
  activeStep: Types.number.isRequired,
  backward: Types.bool,
};

AnimatedLabel.defaultProps = {
  backward: false,
};

export default AnimatedLabel;
