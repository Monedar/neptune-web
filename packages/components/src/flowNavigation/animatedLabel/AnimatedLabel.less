@import (reference) '@transferwise/neptune-css/src/variables/neptune-tokens.less';
@import (reference) '@transferwise/neptune-css/src/less/variables/_typography.less';

@transition-timing-function: ease-in-out;
@transition-duration: 0.3s;
@short-transition-duration: 0.15s;

.np-animated-label {
  @slide-length: 8px;
  position: absolute;
  white-space: nowrap;
  transition: all @transition-duration @transition-timing-function @short-transition-duration;
  left: var(--size-32);
  &-enter {
    opacity: 0;
    transform: translateX(@slide-length);

    &.np-animated-label--backward {
      transform: translateX(-@slide-length);
    }
  }

  &-enter-active {
    opacity: 1;
    transform: translateX(0);

    &.np-animated-label--backward {
      transform: translateX(0);
    }
  }

  &-exit {
    opacity: 1;
    transform: translateX(0);
  }

  &-exit-active {
    opacity: 0;
    transform: translateX(-@slide-length);
    transition: all @transition-duration @transition-timing-function;

    &.np-animated-label--backward {
      transform: translateX(@slide-length);
    }
  }
}
