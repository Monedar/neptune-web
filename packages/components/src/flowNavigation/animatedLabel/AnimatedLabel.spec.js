import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

import AnimatedLabel from '.';

const props = {
  activeStep: 0,
  backward: false,
  labels: ['label1', 'label2'],
};
// jest.mock('react-transition-group/CSSTransition', () => ({ children }) => children);
jest.useFakeTimers();
describe('AnimatedLabel', () => {
  it(`renders no labels`, () => {
    const { container } = render(<AnimatedLabel {...props} />);
    expect(container).toMatchSnapshot();
  });
  it(`renders label 1`, () => {
    const { container } = render(<AnimatedLabel {...props} activeStep={1} />);
    expect(container).toMatchSnapshot();
  });
  it(`renders label 2`, () => {
    const { container } = render(<AnimatedLabel {...props} activeStep={2} />);
    expect(container).toMatchSnapshot();
  });
});
