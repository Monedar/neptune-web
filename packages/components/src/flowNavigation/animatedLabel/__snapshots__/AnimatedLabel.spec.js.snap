// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedLabel renders label 1 1`] = `
<div>
  <div
    aria-hidden="false"
    class="np-animated-label"
  >
    label1
  </div>
</div>
`;

exports[`AnimatedLabel renders label 2 1`] = `
<div>
  <div
    aria-hidden="false"
    class="np-animated-label"
  >
    label2
  </div>
</div>
`;

exports[`AnimatedLabel renders no labels 1`] = `<div />`;
